src/NHibernate/Linq/Visitors/SelectClauseNominator.cs

```diff
@@ -109,6 +112,29 @@ public override Expression VisitExpression(Expression expression)
 			return expression;
 		}
 
+		private bool DbCanCast(System.Type type)
```

Comment:
@hazzik I don't think this is true.
At this point we are trying to nominate expressions that can be converted to HQL.
I don't think that conditionals in where clauses are checking dialect casting capabilities, and yet they all generate a case statement wrapped in a cast statement, using the IDENT node with the type to cast as.

I assume the dialect is responsible for converting the IDENT node into an appropriate DB type. If an IDENT type isn't supported, what does the dialect do?